src/main/java/ch/njol/skript/expressions/ExprExperience.java

```diff
@@ -87,31 +96,41 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 	
 	@Override
 	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		if (!(e instanceof ExperienceSpawnEvent))
+		double d;
+		if (e instanceof ExperienceSpawnEvent)
+			d = ((ExperienceSpawnEvent) e).getSpawnedXP();
+		else if (e instanceof BlockBreakEvent)
+			d = ((BlockBreakEvent) e).getExpToDrop();
+		else
 			return;
-		if (delta == null) {
-			((ExperienceSpawnEvent) e).setSpawnedXP(0);
-			return;
-		}
-		double d = 0;
-		for (final Object o : delta) {
-			final double v = o instanceof Experience ? ((Experience) o).getXP() : ((Number) o).doubleValue();
-			switch (mode) {
-				case ADD:
-				case SET:
-					d += v;
-					break;
-				case REMOVE:
-				case REMOVE_ALL:
-					d -= v;
-					break;
-				case RESET:
-				case DELETE:
-					assert false;
-					break;
+		
+		if (delta != null)
+			for (final Object o : delta) {
+				final double v = o instanceof Experience ? ((Experience) o).getXP() : ((Number) o).doubleValue();
+				switch (mode) {
+					case ADD:
+						d += v;
+						break;
+					case SET:
+						d = v;
+						break;
+					case REMOVE:
+					case REMOVE_ALL:
```

Comment:
I think REMOVE_ALL should be removed here? I don't *think* it's appropriate usage of it.

Reply:
good call, I removed it.

Reply:
needs removed from acceptChange too ;D

Reply:
you're so needy ;) haha, jk jk... ok its been removed.!